internal/project/golang/generate.go

```diff
@@ -312,5 +318,51 @@ func (generate *Generate) CompileDockerfile(output *dockerfile.Output) error {
 		}
 	}
 
+	if generate.versionPackagePath() != "" {
+		output.Stage("embed-generate").From("tools").
+			Step(step.Arg("SHA")).
+			Step(step.Arg("TAG")).
+			Step(step.Arg("NAME")).
+			Step(step.WorkDir("/src")).
+			Step(step.Script(fmt.Sprintf(command, generate.versionPackagePath())))
+
+		output.Stage("embed-abbrev-generate").From("embed-generate").
+			Step(step.WorkDir("/src")).
+			Step(step.Script(fmt.Sprintf(abbrevCommand, generate.versionPackagePath())))
+
+		src := fmt.Sprintf("/src/%s", generate.versionPackagePath())
+
+		generateStage.Step(step.Copy(src, generate.versionPackagePath()).From("embed-abbrev-generate"))
+
+		generate.meta.VersionPackagePath = generate.versionPackagePath()
+	}
+
+	return nil
+}
+
+// CompileTemplates implements [template.Compiler].
+func (generate *Generate) CompileTemplates(output *template.Output) error {
+	if generate.versionPackagePath() == "" {
+		return nil
+	}
+
+	output.Define(fmt.Sprintf("%s/version.go", generate.versionPackagePath()), templates.VersionGo).
+		PreamblePrefix("// ").
+		WithLicense().
+		NoOverwrite()
+
 	return nil
 }
+
+func (generate *Generate) versionPackagePath() string {
+	return strings.TrimSpace(generate.VersionPackagePath)
+}
+
+const (
+	command = `mkdir -p %[1]s/data && \
+    echo -n ${SHA} > %[1]s/data/sha && \
+    echo -n ${TAG} > %[1]s/data/tag`
+
+	abbrevCommand = `echo -n 'undefined' > %[1]s/data/sha && \
+    echo -n 'undefined' > %[1]s/data/tag`
```

Comment:
should the abbrev tag use the actual abbreviated git tag? like in Talos?

Reply:
It will not work for `kres` since it doesn't have tags at the moment, so it prints the commit sha into the file and creates the "cycle between tag file and compilation". I'm not sure about our other projects.

Reply:
we can `make` properly inject it as 'undefined' if there are no tags? 

```shell
$ git describe --tags
fatal: No names found, cannot describe anything.
```

this fails if there are no tags

Reply:
Done using this line `git describe --tags && echo -n ${ABBREV_TAG} > %[1]s/data/tag || echo -n 'undefined' > %[1]s/data/tag`